include/alpaka/core/Cuda.hpp

```diff
@@ -119,10 +119,17 @@ namespace alpaka
     #pragma clang diagnostic push
     #pragma clang diagnostic ignored "-Wmissing-braces"
 #endif
+#if BOOST_COMP_GNUC
+    #pragma GCC diagnostic push
+    #pragma GCC diagnostic ignored "-Wmissing-field-initializers"
+#endif
                     std::array<cudaError_t, sizeof...(ignoredErrorCodes)> const aIgnoredErrorCodes{std::forward<TErrors>(ignoredErrorCodes)...};
```

Comment:
Shouldn't we just write
```C++
std::array<cudaError_t, sizeof...(ignoredErrorCodes)> const aIgnoredErrorCodes{{ignoredErrorCodes...}};
```
?
Source: https://stackoverflow.com/questions/28845419/variadic-template-for-forward-list-initialization-of-stdarray/28845794#28845794